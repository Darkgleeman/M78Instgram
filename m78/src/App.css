.App {
  /* text-align: center; */
  height: 100vh;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  
}

.content {
  width: 100%;
  height: 100%;
}

.content-with-navbar {
  margin-left: 240px;
  width: 100%;
  height: 100%;
}


.navbar-container {
  position: fixed;
  left: 0;
  height: 100%;
  border-right:  1px solid #f2f2f2;
  background-color: #fbfafa;
}

@media screen and (max-width: 740px) {
  .navbar-container{
    display: none;
  }
  
}