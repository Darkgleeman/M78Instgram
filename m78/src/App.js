import React, { useState, useEffect } from 'react';
import {
  BrowserRouter, Routes, Route, Navigate,
} from 'react-router-dom';
import './App.css';
import LoginPage from './components/login/LoginPage';
import Signup from './components/signup/Signup';
import HomePage from './components/homepage/HomePage';
import Navbar from './components/nav/Navbar';
import Profile from './components/profile_page/Profile';
import Followings from './components/follow/Followings';
import Followers from './components/follow/Followers';
import ErrorPage from './components/error_page/ErrorPage';
import BadActionModal from './components/bad_action_modal/BadActionModal';

function App() {
  const [appToken, setAppToken] = useState(() => localStorage.getItem('appToken'));
  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
  const userID = localStorage.getItem('userID');

  useEffect(() => {
    // Handler to call on window resize
    function handleResize() {
      setWindowWidth(window.innerWidth);
    }

    // Set the window width initially
    handleResize();

    // Subscribe to the window resize event
    window.addEventListener('resize', handleResize);

    return () => window.removeEventListener('resize', handleResize);
  }, []);

  const isTokenValid = () => appToken !== null && appToken !== undefined && appToken !== '';
  useEffect(() => {
    if (isTokenValid()) {
      // sessionStorage.setItem('appToken', appToken);
      // addHeadersToken(appToken);
    }
  }, [appToken]);

  // Function to check if the token is valid (not expired)

  // Implement logic to check token validity (e.g., decode JWT and check expiry)
  // For now, this just checks if the token exists

  return (
    <div className="App">
      <BrowserRouter>
        {isTokenValid() && (
          <div className="navbar-container">
            <Navbar />
          </div>
        )}
        <div
          className={
            isTokenValid() && windowWidth >= 740
              ? 'content-with-navbar'
              : 'content'
          }
        >
          <Routes>
            <Route
              path="/"
              element={
                isTokenValid() ? (
                  <HomePage userID={userID} />
                ) : (
                  <Navigate to="/login" replace />
                )
              }
            />
            <Route
              path="/login"
              element={
                !isTokenValid() ? (
                  <LoginPage updateAppToken={setAppToken} />
                ) : (
                  <Navigate to="/home" replace />
                )
              }
            />
            <Route
              path="/signup"
              element={
                !isTokenValid() ? <Signup /> : <Navigate to="/home" replace />
              }
            />
            {userID && isTokenValid() && <Route path="/home" element={<HomePage userID={userID} />} />}
            <Route path="/profile/:userID" element={<Profile />} />
            <Route path="/followers/:userID" element={<Followers />} />
            <Route path="/followings/:userID" element={<Followings />} />
            <Route path='/invalid_token' element={<BadActionModal title="Invalid Token" description="Your token is invalid. Please login again." />} />
            <Route path='/expired_token' element={<BadActionModal title="Expired Token" description="Your token has expired. Please login again" />} />
            <Route path="*" element={<ErrorPage />} />
          </Routes>
        </div>
      </BrowserRouter>
    </div>
  );
}

export default App;
