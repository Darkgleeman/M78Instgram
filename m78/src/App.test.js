/**
 * @jest-environment jsdom
 */

import React from 'react';
// import testing library functions
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'; // for expect(...).toBeInTheDocument()

// Import the component to be tested
import App from './App';

describe('App component', () => {
  test('renders without crashing', () => {
    render(<App />);
  });
  test('Weather app matches snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders Sign Up link', () => {
    const { getByText } = render(<App />);
    const linkElement = getByText(/Sign Up/);
    expect(linkElement).toBeInTheDocument();
  });

  // test('checks if navbar is displayed when user is logged in', () => {
  //   render(<App />);
  //   // Replace the 'login' condition with 'true' to simulate logged-in state
  //   expect(screen.getByText('Navbar')).toBeInTheDocument();
  // });

  // test('checks if content is displayed without navbar on smaller screens', () => {
  //   // Simulate a smaller window width
  //   global.innerWidth = 500;
  //   render(<App />);
  //   // Replace the 'login' condition with 'true' to simulate logged-in state
  //   expect(screen.getByText('Content')).toBeInTheDocument();
  // });

  // test('checks if user is able to log in', async () => {
  //   render(<App />);
  //   // Simulate a login action
  //   userEvent.click(screen.getByText('Login'));
  //   expect(screen.getByText('User Dashboard')).toBeInTheDocument();
  // });

  // Add more tests for other user interactions and functionalities
});
