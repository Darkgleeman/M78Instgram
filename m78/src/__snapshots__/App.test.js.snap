// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Weather app matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="content"
  >
    <div
      className="container"
    >
      <div
        className="item"
        id="two"
      >
        <h1
          className="app-name"
          color="white"
        >
          M78
        </h1>
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root custom-textfield css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="username"
            id="username-label"
          >
            Username
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={true}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="username"
              name="username"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root custom-textfield css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          onKeyDown={[Function]}
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink={false}
            htmlFor="password"
            id="password-label"
          >
            Password
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled={false}
              id="password"
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="password"
            />
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-yjsfm1"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <a
          className="forget-password-prompt"
          href="/login"
          onClick={[Function]}
          style={
            {
              "color": "white",
              "float": "left",
              "marginTop": "10px",
            }
          }
        >
          Forgot your password?
        </a>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1drpt8a-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          Sign In
        </button>
        <div
          style={
            {
              "marginRight": "auto",
            }
          }
        >
          <div>
            Don't have an account? 
            <a
              href="/signup"
              onClick={[Function]}
              style={
                {
                  "color": "white",
                }
              }
            >
               Sign Up! 
            </a>
          </div>
        </div>
      </div>
      <div
        className="login-footer"
      >
        <div
          className="fixed-site-footer"
        >
          <p>
            Every Pixel Tells a Story @M78
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
